lib/sandbox/memory_store.go

```diff
@@ -0,0 +1,92 @@
+package sandbox
+
+import "sync"
+
+// memoryStore implements a Store in memory.
+type memoryStore struct {
+	s map[string]*Sandbox
+	sync.RWMutex
+}
+
+// NewMemoryStore initializes a new memory store.
+func NewMemoryStore() Storer {
+	return &memoryStore{
+		s: make(map[string]*Sandbox),
+	}
+}
+
+// Add appends a new sandbox to the memory store.
+// It overrides the id if it existed before.
+func (c *memoryStore) Add(id string, cont *Sandbox) {
+	c.Lock()
+	c.s[id] = cont
```

Comment:
Golang question.  Is it better to do a defer unlock here (and a couple spots below) just on the extremely small chance that there might be an error before you get to the unlock call?

Reply:
nope, it's better to do this as defer as runtime cost, this is a simple case and there cannot be any error till the call to Unlock

Reply:
OK thanks for the background info, much appreciated.